homeassistant/components/brother/config_flow.py

```diff
@@ -127,6 +136,63 @@ async def async_step_zeroconf_confirm(
             },
         )
 
+    async def async_step_reconfigure(
+        self, _: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        if TYPE_CHECKING:
+            assert entry is not None
+
+        self.host = entry.data[CONF_HOST]
+        self.entry = entry
+
+        return await self.async_step_reconfigure_confirm()
+
+    async def async_step_reconfigure_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        errors = {}
+
+        if TYPE_CHECKING:
+            assert self.entry is not None
+
+        if user_input is not None:
+            try:
+                _, serial = await validate_input(self.hass, user_input)
+            except InvalidHost:
+                errors[CONF_HOST] = "wrong_host"
+            except (ConnectionError, TimeoutError):
+                errors["base"] = "cannot_connect"
+            except SnmpError:
+                errors["base"] = "snmp_error"
+            else:
+                if serial.lower() != self.entry.unique_id:
+                    return self.async_abort(reason="another_device")
+
+                self.hass.config_entries.async_update_entry(
+                    self.entry,
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_TYPE: self.entry.data[CONF_TYPE],
+                    },
+                )
+                await self.hass.config_entries.async_reload(self.entry.entry_id)
+                return self.async_abort(reason="reconfigure_successful")
+
+        return self.async_show_form(
+            step_id="reconfigure_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self.host): str,
+                }
+            ),
```

Comment:
Instead of this, use a constant schema and call `self.add_suggested_values_to_schema` to add the previously configured host as suggestion